wire-ios-request-strategy/Sources/Synchronization/QuickSyncObserverTests.swift

```diff
@@ -0,0 +1,75 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+final class QuickSyncObserverTests: MessagingTestBase {
+
+    func testThatSynchronisationStateIsOnline_thenDontWait() async {
+        // given
+        let (_, quickSyncObserver) = Arrangement(coreDataStack: coreDataStack)
+            .withSynchronizationState(.online)
+            .arrange()
+
+        // then test completes
+        wait(timeout: 0.5) {
+            await quickSyncObserver.waitForQuickSyncToFinish()
+        }
+    }
+
+    func testThatSynchronisationStateIsNotOnline_thenWaitUntilQuickSyncCompletes() throws {
+        // given
+        let (_, quickSyncObserver) = Arrangement(coreDataStack: coreDataStack)
+            .withSynchronizationState(.quickSyncing)
+            .arrange()
+
+        Task {
+            try await Task.sleep(nanoseconds: 250_000_000)
```

Comment:
what are we waiting for?

Reply:
Maybe not necessary I'll double check.

Reply:
It's necessary unfortunately

Reply:
add comment why if it was not done on previous pr

Reply:
Done